bindata/etcd/etcd-snapshot-restore.sh

```diff
@@ -49,24 +79,31 @@ find ${MANIFEST_DIR} \
   -printf '...stopping %P\n' \
   -exec mv {} ${MANIFEST_STOPPED_DIR} \;
 
-# Wait for pods to stop
-sleep 30
+# wait for every static pod container to stop
+wait_for_containers_to_stop "${STATIC_POD_CONTAINERS[@]}"
 
-# //TO DO: verify using crictl that etcd and other pods stopped.
-
-# Remove data dir
-echo "Moving etcd data-dir ${ETCD_DATA_DIR}/member to ${ETCD_DATA_DIR_BACKUP}"
-[ ! -d ${ETCD_DATA_DIR_BACKUP} ]  && mkdir -p ${ETCD_DATA_DIR_BACKUP}
-mv ${ETCD_DATA_DIR}/member ${ETCD_DATA_DIR_BACKUP}/member
-
-# Copy snapshot to backupdir
 if [ ! -d ${ETCD_DATA_DIR_BACKUP} ]; then
   mkdir -p ${ETCD_DATA_DIR_BACKUP}
 fi
-cp -p ${SNAPSHOT_FILE} ${ETCD_DATA_DIR_BACKUP}/snapshot.db
 
-# Copy etcd restore pod yaml
-cp -p ${RESTORE_ETCD_POD_YAML} ${MANIFEST_DIR}/etcd-pod.yaml
+# backup old data-dir
+if [ -d "${ETCD_DATA_DIR}/member" ]; then
+  if [ -d "${ETCD_DATA_DIR_BACKUP}/member" ]; then
+    echo "removing previous backup ${ETCD_DATA_DIR_BACKUP}/member"
+    rm -rf ${ETCD_DATA_DIR_BACKUP}/member
+  fi
+  echo "Moving etcd data-dir ${ETCD_DATA_DIR}/member to ${ETCD_DATA_DIR_BACKUP}"
+  mv ${ETCD_DATA_DIR}/member ${ETCD_DATA_DIR_BACKUP}/
+fi
 
 # Restore static pod resources
 tar -C ${CONFIG_FILE_DIR} -xzf ${BACKUP_FILE} static-pod-resources
+
+# Copy snapshot to backupdir
+cp -p ${SNAPSHOT_FILE} ${ETCD_DATA_DIR_BACKUP}/snapshot.db
+
+echo "starting restore-etcd static pod"
+cp -p ${RESTORE_ETCD_POD_YAML} ${MANIFEST_DIR}/etcd-pod.yaml
+
+# start remaining static pods
+restore_static_pods "${STATIC_POD_LIST[@]}"
```

Comment:
The STATIC_POD_LIST also has etcd-pod.yaml. Doesn't overwrite the recovery file we just copied on L196?

Reply:
yep good catch bad copy and paste.